packages/history/modules/index.js

```diff
@@ -0,0 +1,696 @@
+        // still be salvageable in the pagehide event.
+        // See https://html.spec.whatwg.org/#unloading-documents
+        if (!blockers.length) {
+          window.removeEventListener(BeforeUnloadEventType, promptBeforeUnload);
+        }
+      };
+    }
+  };
+
+  return history;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// HASH
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Hash history stores the location in window.location.hash. This makes
+ * it ideal for situations where you don't want to send the location to
+ * the server for some reason, either because you do cannot configure it
+ * or the URL space is reserved for something else.
+ */
+export function createHashHistory({ window = document.defaultView } = {}) {
+  let globalHistory = window.history;
+
+  function getIndexAndLocation() {
+    let { pathname = '/', search = '', hash = '' } = parsePath(
+      window.location.hash.substr(1)
+    );
+    let state = globalHistory.state || {};
+    return [
+      state.idx,
+      readOnly({
+        pathname,
+        search,
+        hash,
+        state: state.usr || null,
+        key: state.key || 'default'
+      })
+    ];
+  }
+
+  let blockedPopTx = null;
+  function handlePop() {
+    if (blockedPopTx) {
+      blockers.call(blockedPopTx);
+      blockedPopTx = null;
+    } else {
+      let nextAction = PopAction;
+      let [nextIndex, nextLocation] = getIndexAndLocation();
+
+      if (blockers.length) {
+        if (nextIndex != null) {
+          let n = index - nextIndex;
+          if (n) {
+            // Revert the POP
+            blockedPopTx = {
+              action: nextAction,
+              location: nextLocation,
+              retry() {
+                go(n * -1);
+              }
+            };
+
+            go(n);
+          }
+        } else {
+          // Trying to POP to a location with no index. We did not create
+          // this location, so we can't effectively block the navigation.
+          warning(
+            false,
+            // TODO: Write up a doc that explains our blocking strategy in
+            // detail and link to it here so people can understand better
+            // what is going on and how to avoid it.
+            `You are trying to block a POP navigation to a location that was not ` +
+              `created by the history library. The block will fail silently in ` +
+              `production, but in general you should do all navigation with the ` +
+              `history library (instead of using window.history.pushState directly) ` +
+              `to avoid this situation.`
+          );
+        }
+      } else {
+        applyTx(nextAction);
+      }
+    }
+  }
+
+  window.addEventListener(PopStateEventType, handlePop);
+
+  // TODO: Is this still necessary? Which browsers do
+  // not trigger popstate when the hash changes?
```

Comment:
According to the MDN, popstate does not fire on hash changes for IE11 and old (trident) Edge versions.

If you want to support IE11, you have to keep the workaround.

https://developer.mozilla.org/de/docs/Web/API/Window/popstate_event

Reply:
That's what I thought. Thanks for confirming :)